lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -943,4 +943,78 @@ public void detach_success_callback_detached() throws AblyException {
 				ably.close();
 		}
 	}
+
+	/**
+	 * <p>
+	 * Validate publish will result in an error, when the channel is
+	 * in or moves to the FAILED state before the operation succeeds
+	 * </p>
+	 * <p>
+	 * Spec: RTL6c3
+	 * </p>
+	 *
+	 * @throws AblyException
+	 */
+	@Test
```

Comment:
> The only assertions are checking for channel state, not for any exception from publish(). Also, if there is an exception, the finally block will be executed and there are no assertions - the test will just pass.

Any uncatched exception will make our test fail. No worries there.


> This test should call publish(), catch the expected exception, and may also then check the expected channel state.

I am doing that on [line 975](https://github.com/gokhanbarisaker/ably-java/blob/channel-test-implicitattach/lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java#L975)